Sources/LiveViewNative/Views/Controls and Indicators/Pickers/MultiDatePicker.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  MultiDatePicker.swift
+//  LiveViewNative
+//
+//  Created by Shadowfacts on 3/8/23.
+//
+
+#if os(iOS)
+import SwiftUI
+
+struct MultiDatePicker<R: RootRegistry>: View {
+    private let context: LiveContext<R>
+    @ObservedElement private var element
+    @Attribute("start") private var start: Date?
+    @Attribute("end") private var end: Date?
+    @LiveBinding(attribute: "value-binding") private var dates: [SelectedDate] = []
+    @State private var dateComponents = Set<DateComponents>()
+    @State private var skipSettingDateComponents = false
```

Comment:
I think this could be simplified by using a `Set` for the `dates` binding:
```suggestion
    @LiveBinding(attribute: "value-binding") private var dates: Set<SelectedDate> = []
    private var dateComponents: Binding<Set<DateComponents>> {
        Binding {
            Set(self.dates.map(\.dateComponents))
        } set: {
            self.dates = Set($0.map(SelectedDate.init(dateComponents:)))
        }
    }
```
Then you can get rid of the `onChange` modifiers.

Reply:
Huh, it's weird that deselection works with a set-backed LiveBinding but not when I tried with an array, even though though converting to/from Set<DateComponents> was exactly the same.